tiled/_tests/test_directory_walker.py

```diff
@@ -130,18 +143,154 @@ async def test_skip_all_in_combination(tmpdir):
     for i in range(2):
         tifffile.imwrite(Path(tmpdir, "one", f"image{i:05}.tif"), data)
 
-    tree = in_memory()
+    catalog = in_memory(writable_storage=tmpdir)
     # By default, both file and tiff sequence are registered.
-    with Context.from_app(build_app(tree)) as context:
-        await register(tree, tmpdir)
+    with Context.from_app(build_app(catalog)) as context:
+        await register(catalog, tmpdir)
         client = from_context(context)
         assert "a" in client
         assert "a" in client["one"]
         assert "image" in client["one"]
 
     # With skip_all, directories and tiff sequence are registered, but individual files are not
-    with Context.from_app(build_app(tree)) as context:
-        await register(tree, tmpdir, walkers=[tiff_sequence, skip_all])
+    with Context.from_app(build_app(catalog)) as context:
+        await register(catalog, tmpdir, walkers=[group_tiff_sequences, skip_all])
         client = from_context(context)
         assert list(client) == ["one"]
         assert "image" in client["one"]
+
+
+@pytest.mark.asyncio
+async def test_tiff_seq_custom_sorting(tmpdir):
+    "Register TIFFs that are not in alphanumeric order."
+    N = 10
+    ordering = list(range(N))
+    random.Random(0).shuffle(ordering)
+    files = []
+    for i in ordering:
+        file = Path(tmpdir, f"image{i:05}.tif")
+        files.append(file)
+        tifffile.imwrite(file, i * data)
```

Comment:
Because the tests rely on the pixel values, it could be useful to import `data` as `ones_data` or something that clarifies that this is a block of `1`s.